From 163497039957450779672506b126cc2b48d50b4b Mon Sep 17 00:00:00 2001
From: nferre <nferre@50fbe906-d41e-0410-8a96-31537896a350>
Date: Tue, 1 Jun 2010 13:53:55 +0000
Subject: [PATCH] sound: atmel ac97: add warm reset if no reset pin

In case of no reset pin defined in ac97c driver, we issue a warm reset anyway.



git-svn-id: svn://rfolxts01.rfo.atmel.com/at91_sandbox/linux-2.6.x/branches/linux-2.6.30-at91@15227 50fbe906-d41e-0410-8a96-31537896a350
---
 sound/atmel/ac97c.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/sound/atmel/ac97c.c b/sound/atmel/ac97c.c
index 2693433..be1ba32 100644
--- a/sound/atmel/ac97c.c
+++ b/sound/atmel/ac97c.c
@@ -1,3 +1,5 @@
+#define DEBUG 12
+#define VERBOSE_DEBUG 12
 /*
  * Driver for Atmel AC97C
  *
@@ -893,6 +895,10 @@ static void atmel_ac97c_reset(struct atmel_ac97c *chip)
 		/* AC97 v2.2 specifications says minimum 1 us. */
 		udelay(2);
 		gpio_set_value(chip->reset_pin, 1);
+	} else {
+		ac97c_writel(chip, MR, AC97C_MR_WRST | AC97C_MR_ENA);
+		udelay(2);
+		ac97c_writel(chip, MR, AC97C_MR_ENA);
 	}
 }
 
-- 
1.5.6.5

